src/Engine/Data/ShaderProgramManager.hpp

```diff
@@ -112,6 +118,8 @@ class RA_ENGINE_API ShaderProgramManager final
     std::map<std::string,
              std::pair<std::unique_ptr<globjects::File>, std::unique_ptr<globjects::NamedString>>>
         m_namedStrings;
+
+    std::string m_defaultVersion { "#version 440" };
```

Comment:
Default version should be 410 to be supported on macos. 
This default version might be updated at runtime, once opengl context is active, by using `gl::glGetString( gl::GLenum( GL_SHADING_LANGUAGE_VERSION )` and transforming the returned string to remove the dot (e.g. 4.10 --> 410)